File: Admin_p/Admin_p.Web/Modules/electroDB/ShoppingCarts/ShoppingCartsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Admin_p.electroDB.Columns;

[ColumnsScript("electroDB.ShoppingCarts")]
[BasedOnRow(typeof(ShoppingCartsRow), CheckNames = true)]
public class ShoppingCartsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CartId { get; set; }
    public string UserName { get; set; }
    public string ProductName { get; set; }
    public int Quantity { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/ShoppingCartsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ShoppingCartsRow } from "./ShoppingCartsRow";

export interface ShoppingCartsColumns {
    CartId: Column<ShoppingCartsRow>;
    UserName: Column<ShoppingCartsRow>;
    ProductName: Column<ShoppingCartsRow>;
    Quantity: Column<ShoppingCartsRow>;
    CreatedAt: Column<ShoppingCartsRow>;
    UpdatedAt: Column<ShoppingCartsRow>;
}

export class ShoppingCartsColumns extends ColumnsBase<ShoppingCartsRow> {
    static readonly columnsKey = 'electroDB.ShoppingCarts';
    static readonly Fields = fieldsProxy<ShoppingCartsColumns>();
}